src/audio/eq_iir.c

```diff
@@ -413,6 +414,18 @@ static int iir_cmd_set_data(struct comp_dev *dev,
 		break;
 	case SOF_CTRL_CMD_BINARY:
 		trace_eq("sbi");
+
+		if (dev->state != COMP_STATE_READY) {
+			/* It is a valid request but currently this is not
+			 * supported during playback/capture. The driver will
+			 * re-send data in next resume when idle and the new
+			 * EQ configuration will be used when playback/capture
+			 * starts.
+			 */
+			trace_eq("esr");
+			return 0;
```

Comment:
We should return -EBUSY here. That way the driver knows it has to wait until idle.

Reply:
Thanks, this works well! Seems there no need to update driver to handle this.